### DISCUSSION TITLE
client protocol.dart imports nonexistent file when using custom-serializable objects

### DISCUSSION THREAD

**Discussion by jason-downs** [serverpod/serverpod#3068] [Q&A] [answered]
I have created custom serializable classes as described in the docs: https://docs.serverpod.dev/concepts/serialization#custom-serializable-objects

Rather than being an external package, this is a class that is defined in my project's server. I specify the relevant classes in the server's `config/generator.yaml`, using
```
extraClasses:
  - project:src/path/to/file.dart:ClassName
```

Then when I `serverpod generate`, on the client side `protocol.dart`, auto-generates this line:
`package:myproject_client/src/path/to/file.dart`
which doesn't exist and therefore is an error.

The file doesn't exist on the client side, which means it either should have been auto-generated when running `serverpod generate`, or is it supposed to reference the `myproject_server` package instead of client? Not sure but it's impossible to use these custom serializable classes until this is solved.

**Expected behavior**

The client code for these custom-serializable classes should be gererated with `serverpod generate` without errors, or if there is an error it should say so instead of just saying "done" when running `serverpod generate`.

**Serverpod versions**
 -  _CLI Version: 1.1.1
 - _Library version: 1.1.1

**Comment by AndryHTC** [comment]
I have the same problem. The same happens in client.dart when generating endpoints with custom classes in shared packages

**Comment by SandPod** [comment, marked as answer]
In order for the class to be deserialized on the client side, the client needs to be able to access the class definition. In the examples provided, the extra classes are always placed in a separate package so that they can be imported by both the server and the client.

By defining the extra classes import by just a path after `project:`, our generator assumes that this definition is available in both the client and the server at the same path.

If you want to export or use the classes defined in your server, you should be able to replace `project:src/path/to/file.dart:ClassName` with `project:<your_server_project>/<your_exported_library>/file.dart:ClassName`

You can see examples of that type of import here: https://github.com/serverpod/serverpod/blob/0120e776e1af205233f1276960a1cea6f3ddb81e/tests/serverpod_test_server/config/generator.yaml#L11
